core/src/main/java/org/apache/iceberg/UpdateRequirement.java

```diff
@@ -36,6 +36,8 @@ public void validate(TableMetadata base) {
     }
   }
 
+  /** @deprecated will be removed in 1.6.0, use {@link AssertUUID} instead */
+  @Deprecated
   class AssertTableUUID implements UpdateRequirement {
```

Comment:
we can directly rename instead of deprecation since it is package private?

Reply:
unfortunately no, because it's inside an interface and so it's public by default